docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -591,63 +594,119 @@ plugin libraries, composed of:
+
+API function `oe_get_evidence()` supports the values listed below in its
+`format_id` parameter. The output evidence will be prefixed with an
+`oe_attestation_header` if the `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit in its
 `flags` parameter is set.
+* `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION`
+* `OE_FORMAT_UUID_SGX_ECDSA_P256`
+* `OE_FORMAT_UUID_SGX_EPID_LINKABLE`
+* `OE_FORMAT_UUID_SGX_EPID_UNLINKABLE`
+
+API function `oe_verify_evidence()` supports the values listed below in its
+`format_id` parameter.
+* `NULL`:
+  * The input evidence is generated by `oe_get_evidence()`, with
+  the `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit set in its `flags` parameter.
+* `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION`:
+  * The input evidence is generated by `oe_get_evidence()` for format
+  `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION`, with the
+  `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit cleared in its `flags` parameter.
+* `OE_FORMAT_UUID_SGX_ECDSA_P256`:
+  * The input evidence is generated by `oe_get_evidence()` for format
+  `OE_FORMAT_UUID_SGX_ECDSA_P256`, with the
+  `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` bit cleared in its `flags` parameter.
+* `OE_FORMAT_UUID_LEGACY_REPORT_REMOTE`:
+  * The input evidence is an OE report generated by the legacy API function
+  `oe_get_report()` with the `OE_REPORT_FLAGS_REMOTE_ATTESTATION` flag.
+* `OE_FORMAT_UUID_SGX_QUOTE_ECDSA_P256`:
+  * The input evidence is an SGX ECDSA quote generated by the
+  Intel SGX SDK DCAP library, or the quote-ex library with algorithm ID
+  `SGX_QL_ALG_ECDSA_P256`.
+
+The table below shows the structure of the evidence data for all the supported
+SGX format IDs, as generated by an attester plugin or verified by a verifier
+plugin.
 
 | Format ID | Evidence structure  |
 | -- | - |
-| `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION` | `oe_report_header \|\| SGX_report(hash) \|\| custom_claims` |
-| `OE_FORMAT_UUID_SGX_ECDSA_P256` | `oe_report_header \|\| SGX_ECDSA_quote(hash) \|\| custom_claims` |
-| `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT` | `oe_report_header \|\| SGX_ECDSA_quote(custom_claims)` |
-| `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE` | `SGX_ECDSA_quote(custom_claims)` |
-| `OE_FORMAT_UUID_SGX_EPID_LINKABLE` | `SGX_EPID_linkable_quote(custom_claims)` |
-| `OE_FORMAT_UUID_SGX_EPID_UNLINKABLE` | `SGX_EPID_unlinkable_quote(custom_claims)` |
+| `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION` | `[ oe_attestation_header ] \|\| SGX_report(hash) \|\| custom_claims` |
+| `OE_FORMAT_UUID_SGX_ECDSA_P256` | `[ oe_attestation_header ] \|\| SGX_ECDSA_p256_quote(hash) \|\| custom_claims` |
+| `OE_FORMAT_UUID_SGX_EPID_LINKABLE` | `[ oe_attestation_header ] \|\| SGX_EPID_linkable_quote(custom_claims)` |
+| `OE_FORMAT_UUID_SGX_EPID_UNLINKABLE` | `[ oe_attestation_header ] \|\| SGX_EPID_unlinkable_quote(custom_claims)` |
+| `OE_FORMAT_UUID_LEGACY_REPORT_REMOTE` | `oe_report_header (for remote attestation) \|\| SGX_ECDSA_p256_quote(custom_claims)` |
+| `OE_FORMAT_UUID_SGX_QUOTE_ECDSA_P256` | `SGX_ECDSA_p256_quote(custom_claims)` |
```

Comment:
Per the SIG attestation meeting, I thought this was going to be called 
`OE_FORMAT_UUID_RAW_SGX_ECDSA_P256` ... as it stands, I'm not sure that the "quote" modifier accurately distinguishes `OE_FORMAT_UUID_SGX_QUOTE_ECDSA_P256` from `OE_FORMAT_UUID_SGX_ECDSA_P256`

Reply:
Agreed the two names could be hard to distinguish. How about `OE_FORMAT_UUID_RAW_SGX_QUOTE_ECDSA`? "quote" is a widely used SGX term.

Reply:
I can accept `OE_FORMAT_UUID_RAW_SGX_QUOTE_ECDSA`. No P256 for now please.